### PR TITLE
Osn video

### PR DESCRIPTION
**Pull request by EddyGharbi**
### Description
Expose a C++ API for the video context libobs API. The old API will remain until the client fully transitions. For the time being `Video.set()` will also save the video settings in the `basic.ini` config file, this is done in order for the old and new API to still working together.

### Motivation and Context
Nodeobs settings is hard to use and we want to transition into having these settings 100% from the client.

### How Has This Been Tested?
I created automated tests using this new AP, all tests are passing.

### Types of changes
- Tweak (non-breaking change to improve existing functionality)

### Checklist:
- [X] The code has been tested.
- [X] All commit messages are properly formatted and commits squashed where appropriate.

### CODE REVIEW

File: obs-studio-server/source/osn-video.cpp [hunk 1]
```diff
@@ -52,4 +64,199 @@ void osn::Video::GetTotalFrames(
...
+    video.base_width = baseWidth;
+    video.base_height = baseHeight;
+    video.output_width = outputWidth;
+    video.output_height = outputHeight;
+    video.output_format = (video_format)outputFormat;
+    video.colorspace = (video_colorspace)colorspace;
+    video.range = (video_range_type)range;
+    video.scale_type = (obs_scale_type)scaleType;
+    video.adapter = 0;
+	video.gpu_conversion = true;
```
Comment (summeroff) [inline obs-studio-server/source/osn-video.cpp]: only line with tab instead of space at the beginning.

File: obs-studio-client/source/video.cpp [hunk 2]
```diff
@@ -76,3 +78,65 @@ Napi::Value osn::Video::encodedFrames(const Napi::CallbackInfo& info)
...
+	video.Set("range", response[9].value_union.ui32);
+	video.Set("scaleType", response[10].value_union.ui32);
+
+	return video;
+}
+
+void osn::Video::set(const Napi::CallbackInfo& info, const Napi::Value &value)
+{
+	Napi::Object video = value.ToObject();
+	if (!video || !video.IsObject()) {
```
Comment (summeroff) [inline obs-studio-client/source/video.cpp]: IsObject() checked after ToObject()

Comment (EddyGharbi) [reply obs-studio-client/source/video.cpp]: Good catch, fixed

File: obs-studio-server/source/osn-video.cpp (hunk 1 above)
Comment (EddyGharbi) [reply obs-studio-server/source/osn-video.cpp]: Good catch, fixed

**Review by summeroff — APPROVED** [review]

**Pull request merged**